### PR TITLE
Add Plan for builder-web

### PR DESCRIPTION
**Pull request by elliott-davis**
Move current builder-web build from builder-api to its own plan.
Update builder-api-proxy to support new builder-web plan.

tenor-108066620

Signed-off-by: Travis Elliott Davis <edavis@chef.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-api-proxy/config/nginx.conf [hunk 1]
```diff
@@ -61,14 +61,17 @@ http {
                    '"$http_referer" "$http_user_agent" - $http_x_forwarded_for';
 
   upstream backend {
-    server 127.0.0.1:{{bind.http.first.cfg.port}};
```
Comment (reset) [inline components/builder-api-proxy/config/nginx.conf]: We do want to keep the singular backend pointing to localhost. Without it, we end up transferring large amounts of data over the network and end up with the performance issues that we saw on the Thursday afternoon/morning in Tokyo after that deploy. You were sitting next to me when I fixed this ;)

File: components/builder-api-proxy/config/nginx.conf [hunk 2]
```diff
@@ -102,6 +105,7 @@ http {
 
     location /habitat.conf.js {
       add_header Cache-Control "private, no-cache, no-store";
+      alias /hab/svc/builder-web/config/;
```
Comment (reset) [inline components/builder-api-proxy/config/nginx.conf]: Nice!

File: components/builder-api/habitat/plan.sh [hunk 3]
```diff
@@ -5,43 +5,14 @@ pkg_maintainer="Jamie Winsor <reset@chef.io>"
...
+core/rust core/gcc core/git core/pkg-config)
 pkg_exports=(
   [port]=http.port
-  [url]=web.app_url
 )
 pkg_exposes=(port)
 pkg_binds=(
   [router]="port"
 )
-bin="bldr-api"
```
Comment (reset) [inline components/builder-api/habitat/plan.sh]: This is required by the `builder-base-plan.sh` file

File: components/builder-web/habitat/default.toml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+events_url            = "https://events.chef.io/events/categories/habitat/"
+roadmap_url           = "https://ext.prodpad.com/ext/roadmap/d2938aed0d0ad1dd62669583e108357efd53b3a6"
+feature_requests_url  = "https://portal.prodpad.com/24539"
+slack_url             = "http://slack.habitat.sh/"
+youtube_url           = "https://www.youtube.com/playlist?list=PL11cZfNdwNyOxlvI1Kq6ae8eVBl5S3IKk"
+demo_app_url          = "https://www.habitat.sh/demo"
+github_app_url        = "https://github.com/apps/habitat-builder"
+
+[web_feature_flags]
+project = true
```
Comment (reset) [inline components/builder-web/habitat/default.toml]: We can probably just call this feature flags now that it's obviously part of web

File: components/builder-web/habitat/plan.sh [hunk 5]
```diff
@@ -0,0 +1,58 @@
+source "../../../support/ci/builder-base-plan.sh"
```
Comment (reset) [inline components/builder-web/habitat/plan.sh]: Don't think you need to include this, it's basically a shitty Rust scaffolding

**Comment by reset** [comment]
tenor-194406151

**Pull request merged**